File: apps/frontend/src/utils/api-client/interfaces/tag.ts
import { UUID } from 'crypto';



export interface TagCreate extends TagBase {}

export interface TagUpdate extends Partial<TagBase> {}

export interface Tag extends TagBase {
  id: UUID;
  created_at: string;
  updated_at: string;
}

export interface TagAssignment {
  entity_id: UUID;
  entity_type: EntityType;
}

File: apps/backend/src/rhesis/backend/app/schemas/tag.py
from enum import Enum
from typing import Optional

from pydantic import UUID4

from rhesis.backend.app.schemas import Base
from rhesis.backend.app.constants import EntityType


# Tag schemas
class TagBase(Base):
    name: str
    icon_unicode: Optional[str] = None
    organization_id: Optional[UUID4] = None
    user_id: Optional[UUID4] = None


class TagCreate(TagBase):
    pass


class TagUpdate(TagBase):
    name: Optional[str] = None


class TagAssignment(Base):
    entity_id: UUID4
    entity_type: EntityType


class TagRead(Base):
    id: UUID4
    name: str
    icon_unicode: Optional[str] = None


class Tag(TagBase):
    pass
